### PR TITLE
Upgrade phpunit & fix tests

### PR DESCRIPTION
**Pull request by antoine-pous**
- [x] Update to PHPUnit 6
- - [x] RouteTest
- - [x] ConfigTest
- - [x] RouterTest
- - [x] composer.json
- [x] Travis-CI
- - [x] Add PHP 7.1
- - [x] Remove PHP version below 7.0
- [x] Validate tests
- - [x] ConfigTest 
- - [x] RouteTest
- - [x] RouterTest
- [x] Config optimizations
- - [x] Improved YAML format
- - [x] Implement JSON support

### CODE REVIEW

**Comment by antoine-pous** [comment]
I need your advice on `testParamsWithoutDynamicFilterMatch`. Currently the test fail on 
```PHP
self::assertEquals(
    array(array('filename' => 'someJsFile.min')),
    $router->match('/js/someJsFile.min.js')->getParameters()
);

self::assertEquals(
    array(array('filename' => 'someJsFile.min.js')),
    $router->match('/js/someJsFile.min.js.js')->getParameters()
);
```

This is due to unsupported multiple matches.
`someJsFile` and `someJsFile.min` are matched and we have multiple choices about this behavior:

1. thrown an exception and force developpers to be more strict
1. always return the last match
1. support this kind of behavior

File: composer.json [hunk 1]
```diff
@@ -21,11 +21,13 @@
...
     "php": "^5.3.0|^7.0",
-    "fig/http-message-util": "^1.1"
+    "fig/http-message-util": "^1.1",
+    "symfony/yaml": "^3.2",
+    "phpunit/php-code-coverage": "^5.0"
   },
   "require-dev": {
     "malukenho/docheader": "^0.1.5",
-    "phpunit/phpunit": "4.8.*",
+    "phpunit/phpunit": "6.*",
```
Comment (malukenho) [inline composer.json]: use `^`

File: src/Route.php [hunk 2]
```diff
@@ -169,8 +181,8 @@ private function substituteFilter($matches)
     private function validateFilters()
     {
         foreach($this->filters as $key => $reg) {
-            if(!preg_match('~^:([[a-z]])$~i', $key)) {
-                throw new Exception('Invalid filter name `'.$key.'` it should contains only letters and start with `:`');
+            if(!preg_match('~^:([a-z_]+)$~i', $key)) {
+                throw new Exception('Invalid filter name `'.$key.'` it should contains only letters, underscore and start with `:`');
```
Comment (malukenho) [inline src/Route.php]: We can throw an specific exception here?

File: src/Route.php [hunk 3]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
         }
 
-        $this->action = !empty($action[1]) && trim($action[1]) !== '' ? $action[1] : null;
-
-        if (!is_null($this->action)) {
-            $instance = new $action[0];
-            call_user_func_array(array($instance, $this->action), $this->parameters);
+        if($action === '__construct') {
+            $instance = new $controller($this->parameters);
         } else {
```
Comment (malukenho) [inline src/Route.php]: can we remove the else?

File: src/Route.php [hunk 4]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
         if ($this->parametersByName) {
             $this->parameters = array($this->parameters);
         }
 
-        $this->action = !empty($action[1]) && trim($action[1]) !== '' ? $action[1] : null;
-
-        if (!is_null($this->action)) {
-            $instance = new $action[0];
-            call_user_func_array(array($instance, $this->action), $this->parameters);
+        if($action === '__construct') {
```
Comment (malukenho) [inline src/Route.php]: code style

File: src/Route.php [hunk 5]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
+            $instance = new $controller($this->parameters);
```
Comment (malukenho) [inline src/Route.php]: `class_exists`?

File: src/Route.php [hunk 6]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
         } else {
-            $instance = new $action[0]($this->parameters);
+            $instance = new $controller;
+            call_user_func_array(array($instance, $action), $this->parameters);
```
Comment (malukenho) [inline src/Route.php]: shall we use `[]`? Also, Why not drop support to php 5.*?

File: src/Route.php [hunk 7]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
         }
     }
 
     public function getAction()
     {
+        $action = explode('::', $this->config['_controller']);
```
Comment (malukenho) [inline src/Route.php]: move `::` to a constant.

File: src/Route.php [hunk 8]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
+
+        $this->controller = $action[0];
```
Comment (malukenho) [inline src/Route.php]: We probably should abstract or use any other package to instantiate objects for us. Look at the doctrine one.

File: src/Route.php [hunk 9]
```diff
@@ -187,24 +199,34 @@ public function setParameters(array $parameters)
...
+        $this->action = !empty($action[1]) && trim($action[1]) !== '' ? $action[1] : '__construct';
+
         return $this->action;
     }
+
+    public function getController()
+    {
+        $this->getAction();
```
Comment (malukenho) [inline src/Route.php]: ouch!

**Review by malukenho — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
As for your question, it should stop on the first assertion and execute controller/method form there.

File: src/Route.php (hunk 6 above)
Comment (antoine-pous) [inline src/Route.php]: See #105

File: src/Route.php (hunk 5 above)
Comment (antoine-pous) [inline src/Route.php]: Why we should test if the class exists? This is note the job of PHPRouter...

Comment (malukenho) [inline src/Route.php]: It's not the job of a router instantiate a class and call it a method too... If that's the point just use a service container or some sort.

File: src/Route.php (hunk 2 above)
Comment (antoine-pous) [inline src/Route.php]: Wich exception would you throw exactly?

File: composer.json (hunk 1 above)
Comment (antoine-pous) [inline composer.json]: Fixed

File: src/Route.php (hunk 2 above)
Comment (malukenho) [inline src/Route.php]: `InvalidArgumentException` or `RuntimeException` probably.

**Comment by antoine-pous** [comment]
@malukenho @dannyvankooten i don't understand why `$router->match()` use this:

```php
$currentDir = dirname($_SERVER['SCRIPT_NAME']);
//...
if ('/' !== $currentDir) {
    $requestUrl = str_replace($currentDir, '', $requestUrl);
}
```

When we use filter regex for JS file like `([[:alnum:].]+).js` on route `/js/:filename` the match is broken. This is because `dirname($_SERVER['SCRIPT_NAME']);` return `.`.

I really don't find any usage where this replacement is necessary, can you give me an example please?

File: src/Route.php (hunk 9 above)
Comment (antoine-pous) [inline src/Route.php]: Fixed

File: src/Route.php (hunk 8 above)
Comment (antoine-pous) [inline src/Route.php]: I've added ne methods, with them PHPRouter is more flexible

**Comment by antoine-pous** [comment]
All tests seem OK.

@malukenho Docheader dependency broke the last travis tests, see https://github.com/malukenho/docheader/issues/17 for more informations.

**Comment by malukenho** [comment]
@antoine-pous As far as I can see your tests is failing due multiples issues.

First, `docheader` doesn't work with php versions bellow `5.5`. So that explains failures for version `5.3` and `5.4`.

Other versions are failing as unit tests is broken, that indicates that bugs were introduced. 
If you're having trouble with `docheader` on your machine. Check if dependencies are installed.

File: .travis.yml [hunk 10]
```diff
@@ -7,28 +7,32 @@ cache:
     - $HOME/.composer/cache
 
 php:
-  - 5.3
-  - 5.4
-  - 5.5
   - 5.6
```
Comment (malukenho) [inline .travis.yml]: Can we drop support to 5.6?

File: .travis.yml [hunk 11]
```diff
@@ -7,28 +7,32 @@ cache:
...
   - 7.0
```
Comment (malukenho) [inline .travis.yml]: I'd also drop support to `7.0`

File: composer.json [hunk 12]
```diff
@@ -1,36 +1,43 @@
...
-  "require-dev": {
-    "malukenho/docheader": "^0.1.5",
-    "phpunit/phpunit": "4.8.*",
-    "squizlabs/php_codesniffer": "2.*"
-  },
-  "autoload": {
-    "psr-4": {
-      "PHPRouter\\": "src/"
+    "require": {
+        "php": "^5.6|^7.0|^7.1",
```
Comment (malukenho) [inline composer.json]: just `^7.1`

File: composer.json [hunk 13]
```diff
@@ -1,36 +1,43 @@
...
+    "authors": [
+        {
+            "name": "Danny van Kooten",
+            "email": "dannyvankooten@gmail.com",
+            "homepage": "https://dannyvankooten.com/"
+        },
+        {
+            "name": "Jefersson Nathan",
+            "email": "malukenho@phpse.net"
+        }
```
Comment (malukenho) [inline composer.json]: Put your name is here.

File: composer.json [hunk 14]
```diff
@@ -1,36 +1,43 @@
...
-      "PHPRouter\\": "src/"
+    "require": {
+        "php": "^5.6|^7.0|^7.1",
+        "fig/http-message-util": "^1.1",
+        "symfony/yaml": "^3.2",
+        "phpunit/php-code-coverage": "4.0.6"
+    },
+    "require-dev": {
+        "malukenho/docheader": "^0.1.5",
+        "phpunit/phpunit": "~5.0",
```
Comment (malukenho) [inline composer.json]: We can use phpunit `^6.0` 😄

File: src/Route.php [hunk 15]
```diff
@@ -55,13 +73,25 @@ class Route
...
      * @var array
      */
-    private $filters = array();
+    private $filters = [];
+
+    /**
+     * Regex used to validate filtername
+     * @var string
+     */
+    private $filtersRegex = ':([a-zA-Z_]+)';
```
Comment (malukenho) [inline src/Route.php]: use a constant.

File: tests/src/PHPRouterTest/RouterTest.php [hunk 16]
```diff
@@ -158,6 +179,23 @@ public function testGenerateNotExistent()
...
 
     /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testWrongFiltername()
+    {
+        $collection = new RouteCollection();
+        $route = new Route(
+            '/user/:user_id',
+            array(
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouterTest.php]: `[]`

File: tests/src/PHPRouterTest/RouterTest.php [hunk 17]
```diff
@@ -134,6 +134,27 @@ public function testParamsWithDynamicFilterMatch()
         );
     }
 
+    public function testCustomParameters()
+    {
+        $collection = new RouteCollection();
+        $route = new Route(
+            '/test/params',
+            array(
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouterTest.php]: `[]`

File: phpunit.xml.dist [hunk 18]
```diff
@@ -1,35 +1,29 @@
 <?xml version="1.0" encoding="UTF-8"?>
-
-<phpunit
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/4.8/phpunit.xsd"
-         backupGlobals="true"
+<phpunit bootstrap="tests/Bootstrap.php"
```
Comment (malukenho) [inline phpunit.xml.dist]: remove bootstrap, just use `vendor/autoload.php`

File: phpunit.xml.dist [hunk 19]
```diff
@@ -1,35 +1,29 @@
...
+         backupGlobals="false"
```
Comment (malukenho) [inline phpunit.xml.dist]: It's `false` by default right?

File: phpunit.xml.dist [hunk 20]
```diff
@@ -1,35 +1,29 @@
...
          backupStaticAttributes="false"
          colors="true"
+         verbose="true"
          convertErrorsToExceptions="true"
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
-         bootstrap="tests/Bootstrap.php">
-
+         processIsolation="false"
```
Comment (malukenho) [inline phpunit.xml.dist]: Isn't it `false` by default?

File: phpunit.xml.dist [hunk 21]
```diff
@@ -1,35 +1,29 @@
...
+         stopOnFailure="false">
```
Comment (malukenho) [inline phpunit.xml.dist]: Isn't it `false` by default?

File: phpunit.xml.dist [hunk 22]
```diff
@@ -1,35 +1,29 @@
...
     <testsuites>
-        <testsuite name="PHP Router">
-            <directory>tests</directory>
+        <testsuite name="PHPRouter Test Suite">
+            <directory>tests/src/PHPRouterTest/</directory>
```
Comment (malukenho) [inline phpunit.xml.dist]: leave it as `tests/` only

File: phpunit.xml.dist [hunk 23]
```diff
@@ -1,35 +1,29 @@
...
-                <directory>./vendor</directory>
-            </exclude>
+            <directory suffix=".php">src/</directory>
         </whitelist>
     </filter>
-
     <logging>
-        <log type="coverage-text" target="php://stdout" showUncoveredFiles="true"/>
+        <log type="tap" target="build/report.tap"/>
+        <log type="junit" target="build/report.junit.xml"/>
```
Comment (malukenho) [inline phpunit.xml.dist]: `junit`?

File: phpunit.xml.dist [hunk 24]
```diff
@@ -1,35 +1,29 @@
...
-                <directory>./tests</directory>
-                <directory>./vendor</directory>
-            </exclude>
+            <directory suffix=".php">src/</directory>
         </whitelist>
     </filter>
-
     <logging>
-        <log type="coverage-text" target="php://stdout" showUncoveredFiles="true"/>
+        <log type="tap" target="build/report.tap"/>
```
Comment (malukenho) [inline phpunit.xml.dist]: are we using logs for some reason?

File: src/Route.php [hunk 25]
```diff
@@ -18,7 +18,7 @@
 namespace PHPRouter;
 
 use Fig\Http\Message\RequestMethodInterface;
-use Exception;
+use InvalidArgumentException;
```
Comment (malukenho) [inline src/Route.php]: Should be on our own namespace.

File: src/Route.php [hunk 26]
```diff
@@ -29,15 +29,33 @@ class Route
     private $url;
 
     /**
+     * Separator used between class and action
+     * @const string
+     */
+    const ACTION_SEPARATOR = '::';
```
Comment (malukenho) [inline src/Route.php]: `Paamayim Nekudotayim`

File: src/Route.php [hunk 27]
```diff
@@ -29,15 +29,33 @@ class Route
...
+    private $methods = [
         RequestMethodInterface::METHOD_GET,
         RequestMethodInterface::METHOD_POST,
         RequestMethodInterface::METHOD_PUT,
         RequestMethodInterface::METHOD_DELETE,
-    );
+    ];
+
+    /**
+     * Controller class
```
Comment (malukenho) [inline src/Route.php]: remove this kind of comments that doesn't bring any clarification. instead rename the property to `$controllerClassName` or similar

File: src/Route.php [hunk 28]
```diff
@@ -29,15 +29,33 @@ class Route
...
+     * @var string
+     */
+    private $controller;
+
+    /**
+    * Method called into the controller class
```
Comment (malukenho) [inline src/Route.php]: same as above

File: src/Route.php [hunk 29]
```diff
@@ -81,12 +111,16 @@ class Route
...
     {
-        $this->url        = $resource;
-        $this->config     = $config;
-        $this->methods    = isset($config['methods']) ? (array) $config['methods'] : array();
-        $this->target     = isset($config['target']) ? $config['target'] : null;
-        $this->name       = isset($config['name']) ? $config['name'] : null;
-        $this->parameters = isset($config['parameters']) ? $config['parameters'] : array();
+        $this->url          = $resource;
+        $this->config       = $config;
+        $this->methods      = isset($config['methods']) ? (array) $config['methods'] : [];
```
Comment (malukenho) [inline src/Route.php]: use `??`

File: src/Route.php [hunk 30]
```diff
@@ -81,12 +111,16 @@ class Route
...
+        $this->target       = isset($config['target']) ? $config['target'] : null;
```
Comment (malukenho) [inline src/Route.php]: use `??`

File: src/Route.php [hunk 31]
```diff
@@ -81,12 +111,16 @@ class Route
...
+        $this->name         = isset($config['name']) ? $config['name'] : null;
```
Comment (malukenho) [inline src/Route.php]: use `??`

File: src/Route.php [hunk 32]
```diff
@@ -81,12 +111,16 @@ class Route
...
+        $this->customParams = isset($config['parameters']) ? (array) $config['parameters'] : [];
```
Comment (malukenho) [inline src/Route.php]: use `??`

File: src/Route.php [hunk 33]
```diff
@@ -81,12 +111,16 @@ class Route
...
+
+        $action = explode(self::ACTION_SEPARATOR, $this->config['_controller']);
+        $this->controller = $action[0];
+        $this->action = !empty($action[1]) && trim($action[1]) !== '' ? $action[1] : '__construct';
```
Comment (malukenho) [inline src/Route.php]: what if class doesn't have a `__construct` method defined?
It'll crash badly. 

https://3v4l.org/qTm0F

File: src/Route.php [hunk 34]
```diff
@@ -182,29 +216,98 @@ public function getParameters()
...
-        }
+    /**
+     * Set action
+     *
+     * @author Antoine Pous
+     * @since 1.3.0
+     * @param string $action Action to call
+     * @return null
+     */
+    public function setAction($action)
```
Comment (malukenho) [inline src/Route.php]: We should avoid setters. please remove them.

File: src/Route.php [hunk 35]
```diff
@@ -182,29 +216,98 @@ public function getParameters()
...
+    }
+
+    /**
+     * Get filter regular expression
+     *
+     * @author Antoine Pous
+     * @since 1.3.0
+     * @return string The regex used to match filters names
+     */
+    public function getFiltersRegex()
```
Comment (malukenho) [inline src/Route.php]: Nope

File: src/Route.php [hunk 36]
```diff
@@ -182,29 +216,98 @@ public function getParameters()
...
+
+    /**
+     * Set filter regular expression
+     *
+     * @author Antoine Pous
+     * @since 1.3.0
+     * @param string The new regex used to match filters names
+     * @return null
+     */
+    public function setFiltersRegex($regex)
```
Comment (malukenho) [inline src/Route.php]: nope

File: src/Router.php [hunk 37]
```diff
@@ -109,25 +109,24 @@ public function match($requestUrl, $requestMethod = RequestMethodInterface::METH
                 continue;
             }
 
-            if ('/' !== $currentDir) {
+            if (!in_array($currentDir, ['.', '/'])) {
```
Comment (malukenho) [inline src/Router.php]: missing third parameter

File: src/Router.php [hunk 38]
```diff
@@ -109,25 +109,24 @@ public function match($requestUrl, $requestMethod = RequestMethodInterface::METH
...
                 $requestUrl = str_replace($currentDir, '', $requestUrl);
             }
 
             $route = rtrim($routes->getRegex(), '/');
-            $pattern = '@^' . preg_quote($this->basePath) . $route . '/?$@i';
+            $pattern = '@^' . $this->basePath . $route . '/?$@i';
```
Comment (malukenho) [inline src/Router.php]: Missing `preg_quote`

File: src/Router.php [hunk 39]
```diff
@@ -109,25 +109,24 @@ public function match($requestUrl, $requestMethod = RequestMethodInterface::METH
...
             if (!preg_match($pattern, $requestUrl, $matches)) {
                 continue;
             }
 
-            $params = array();
+            $params = [];
 
-            if (preg_match_all('/:([\w-%]+)/', $routes->getUrl(), $argument_keys)) {
+            if (preg_match_all('@' . $routes->getFiltersRegex() . '@', $routes->getUrl(), $argument_keys)) {
```
Comment (malukenho) [inline src/Router.php]: the wrapper is very inconsistency across the code

File: tests/src/PHPRouterTest/RouteTest.php [hunk 40]
```diff
@@ -37,6 +38,18 @@ protected function setUp()
                 'name' => 'page'
             )
         );
+
+        $this->routeWithoutAction = new Route(
+            '/page/:page_id',
+            array(
+                '_controller' => 'PHPRouter\Test\SomeController',
+                'methods' => array(
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouteTest.php]: `[]`

File: tests/src/PHPRouterTest/RouteTest.php [hunk 41]
```diff
@@ -37,6 +38,18 @@ protected function setUp()
                 'name' => 'page'
             )
         );
+
+        $this->routeWithoutAction = new Route(
+            '/page/:page_id',
+            array(
+                '_controller' => 'PHPRouter\Test\SomeController',
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouteTest.php]: `::class`

File: tests/src/PHPRouterTest/RouteTest.php [hunk 42]
```diff
@@ -92,5 +105,35 @@ public function testSetName()
     public function testGetAction()
     {
         self::assertEquals('page', $this->routeWithParameters->getAction());
+        self::assertEquals('__construct', $this->routeWithoutAction->getAction());
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouteTest.php]: nope

File: tests/src/PHPRouterTest/RouterTest.php [hunk 43]
```diff
@@ -158,6 +179,23 @@ public function testGenerateNotExistent()
...
+        $collection = new RouteCollection();
+        $route = new Route(
+            '/user/:user_id',
+            array(
+                '_controller' => 'PHPRouter\Test\SomeController::dynamicFilterUrlMatch',
+                'methods' => 'GET',
+            )
+        );
+        $route->setFiltersRegex(':([a-z]+):');
+        $route->setFilters(array(':filename' => '([[:alnum:].]+).js'), true);
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouterTest.php]: `[]`

File: tests/src/PHPRouterTest/RouterTest.php [hunk 44]
```diff
@@ -108,13 +108,13 @@ public function testParamsWithDynamicFilterMatch()
     {
         $collection = new RouteCollection();
         $route = new Route(
-            '/js/:filename.js',
+            '/js/:filename',
             array(
                 '_controller' => 'PHPRouter\Test\SomeController::dynamicFilterUrlMatch',
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouterTest.php]: `::class`

File: tests/src/PHPRouterTest/RouterTest.php [hunk 45]
```diff
@@ -94,7 +94,7 @@ public function testGetParamsInsideControllerMethod()
                 'methods' => 'GET'
             )
         );
-        $route->setFilters(array('page_id' => '([a-zA-Z]+)'), true);
+        $route->setFilters(array(':page_id' => '([a-zA-Z]+)'), true);
```
Comment (malukenho) [inline tests/src/PHPRouterTest/RouterTest.php]: `[]`

File: tests/src/PHPRouterTest/ConfigTest.php [hunk 46]
```diff
@@ -18,23 +18,21 @@
...
     /**
      * @throws \InvalidArgumentException
      */
     public function testConfigThrowsErrorWithWrongParameter()
     {
-        $this->setExpectedException(
-            'InvalidArgumentException',
-            'The file fileNotExisting not exists!'
-        );
+        $this->expectException('InvalidArgumentException');
```
Comment (malukenho) [inline tests/src/PHPRouterTest/ConfigTest.php]: `::class`

File: tests/src/PHPRouterTest/ConfigTest.php [hunk 47]
```diff
@@ -18,23 +18,21 @@
...
 use PHPRouter\Config;
-use PHPUnit_Framework_TestCase;
+use PHPUnit\Framework\TestCase;
 
 /**
  * @author Jefersson Nathan <malukenho@phpse.net>
  * @package PHPRouter\Test
  */
-class ConfigTest extends PHPUnit_Framework_TestCase
+class ConfigTest extends TestCase
```
Comment (malukenho) [inline tests/src/PHPRouterTest/ConfigTest.php]: `final`

File: src/Route.php [hunk 48]
```diff
@@ -182,29 +216,98 @@ public function getParameters()
 
     public function setParameters(array $parameters)
```
Comment (malukenho) [inline src/Route.php]: can we remove it?

File: .travis.yml [hunk 49]
```diff
@@ -7,28 +7,32 @@ cache:
...
 before_script:
-    - composer self-update
-    - composer install
-    - mkdir -p build/logs
+  - travis_retry composer update ${COMPOSER_FLAGS} --no-interaction --prefer-dist
 
 script:
-    - vendor/bin/phpunit -c phpunit.xml.dist
-    - vendor/bin/docheader check src/ tests/
+  - vendor/bin/phpunit -c phpunit.xml.dist
```
Comment (malukenho) [inline .travis.yml]: use the call defined on `composer.json`

File: .travis.yml [hunk 50]
```diff
@@ -7,28 +7,32 @@ cache:
...
+  - vendor/bin/docheader check src/ tests/
```
Comment (malukenho) [inline .travis.yml]: use the call defined on `composer.json`

File: .travis.yml [hunk 51]
```diff
@@ -7,28 +7,32 @@ cache:
...
+  - vendor/bin/phpcs --standard=psr2 src/
```
Comment (malukenho) [inline .travis.yml]: use the call defined on `composer.json`

File: .travis.yml [hunk 52]
```diff
@@ -7,28 +7,32 @@ cache:
...
+  - vendor/bin/phpunit --coverage-text --coverage-clover=coverage.clover
```
Comment (malukenho) [inline .travis.yml]: use the call defined on `composer.json`

File: composer.json [hunk 53]
```diff
@@ -1,36 +1,43 @@
...
-    "rest"
-  ],
-  "homepage": "https://github.com/dannyvankooten/PHP-Router",
-  "license": "MIT",
-  "authors": [
-    {
-      "name": "Danny van Kooten",
-      "email": "dannyvankooten@gmail.com",
-      "homepage": "http://dannyvankooten.com/"
+    "name": "dannyvankooten/php-router",
```
Comment (malukenho) [inline composer.json]: Why `retab` this file?

**Review by malukenho — CHANGES_REQUESTED** [review]

**Comment by malukenho** [comment]
@antoine-pous if we are bring new things to this code, and refactor this code as you wish, may consider follow `psr-7`  and `psr-11`.

File: .travis.yml (hunk 10 above)
Comment (antoine-pous) [inline .travis.yml]: PHP 5.6 is under security support until 31 Dec 2018

File: .travis.yml (hunk 11 above)
Comment (antoine-pous) [inline .travis.yml]: PHP 7.0 is under active support until 3 Dec 2017 and security support until 3 Dec 2018

File: composer.json (hunk 12 above)
Comment (antoine-pous) [inline composer.json]: See previous answers

File: composer.json (hunk 14 above)
Comment (antoine-pous) [inline composer.json]: Not currently, due to the support of PHP5.6, but we can creat a branch for each major version of PHP no? :)

File: src/Route.php (hunk 26 above)
Comment (antoine-pous) [inline src/Route.php]: Oh! You're right :confused:

File: src/Route.php (hunk 27 above)
Comment (antoine-pous) [inline src/Route.php]: Fixed

File: src/Router.php (hunk 37 above)
Comment (antoine-pous) [inline src/Router.php]: The third parameter is optionnal. Should we really fill in all the optional parameters?

File: phpunit.xml.dist (hunk 23 above)
Comment (antoine-pous) [inline phpunit.xml.dist]: oops my bad, i took phpunit config from my another project wich contain some JAVA :man_facepalming:

File: phpunit.xml.dist (hunk 24 above)
Comment (antoine-pous) [inline phpunit.xml.dist]: Personally i use them. I find the log easier to read and interpret than the terminal.

File: phpunit.xml.dist (hunk 18 above)
Comment (antoine-pous) [inline phpunit.xml.dist]: The tests crash with this value

File: phpunit.xml.dist (hunk 19 above)
Comment (antoine-pous) [inline phpunit.xml.dist]: PHPUnit author made regulary changes, i want avoid problems due to configuration.

File: src/Route.php (hunk 36 above)
Comment (antoine-pous) [inline src/Route.php]: I think we should let the developper decide wich is the regex used to validate filternames. Sometime it's usefull to use the same syntaxe from another major component of our project.

If it broke the router this is not our problem, we just give a method to set custom filtername validation.

File: src/Route.php (hunk 48 above)
Comment (antoine-pous) [inline src/Route.php]: Router use it :/

File: src/Route.php (hunk 34 above)
Comment (antoine-pous) [inline src/Route.php]: Why?

File: src/Router.php (hunk 37 above)
Comment (malukenho) [inline src/Router.php]: Yep. Be strict at types.

File: phpunit.xml.dist (hunk 24 above)
Comment (malukenho) [inline phpunit.xml.dist]: If it's not part of the `ci` we can remove it. also, you can use a local `phpunit.xml` with your own configs. Just leave what's relevant to the project here.

File: phpunit.xml.dist (hunk 19 above)
Comment (malukenho) [inline phpunit.xml.dist]: Just lock on a specifically version

File: phpunit.xml.dist (hunk 18 above)
Comment (malukenho) [inline phpunit.xml.dist]: You need to register the namespace using `autoload-dev`

**Pull request closed without merging**